pkg/services/ngalert/api/authorization.go

```diff
@@ -75,43 +49,43 @@ var (
 			Version:     1,
 			Permissions: ac.ConcatPermissions(alertingReaderRole.Role.Permissions, []ac.Permission{
 				{
-					Action: ActionAlertingRuleCreate,
-					Scope:  ScopeNamespaceAll,
+					Action: ac.ActionAlertingRuleCreate,
+					Scope:  dashboards.ScopeFoldersAll,
 				},
 				{
-					Action: ActionAlertingRuleUpdate,
-					Scope:  ScopeNamespaceAll,
+					Action: ac.ActionAlertingRuleUpdate,
+					Scope:  dashboards.ScopeFoldersAll,
 				},
 				{
-					Action: ActionAlertingRuleDelete,
-					Scope:  ScopeNamespaceAll,
+					Action: ac.ActionAlertingRuleDelete,
+					Scope:  dashboards.ScopeFoldersAll,
 				},
 				{
-					Action: ActionAlertingRuleCreate,
-					Scope:  ScopeDatasourcesAll,
+					Action: ac.ActionAlertingRuleCreate,
+					Scope:  datasources.ScopeDatasourcesAll,
 				},
 				{
-					Action: ActionAlertingRuleUpdate,
-					Scope:  ScopeDatasourcesAll,
+					Action: ac.ActionAlertingRuleUpdate,
+					Scope:  datasources.ScopeDatasourcesAll,
 				},
 				{
-					Action: ActionAlertingRuleDelete,
-					Scope:  ScopeDatasourcesAll,
+					Action: ac.ActionAlertingRuleDelete,
+					Scope:  datasources.ScopeDatasourcesAll,
```

Comment:
question (not related to this PR): I didn't realise that we also had introduced datasource scopes to alerting. Is the plan to use folder and datasource scopes interchangeably?

Reply:
Grafana supports two alerting APIs: Grafana Alerting - in-process alerting engine, external providers (Prometheus, Loki, Cortex, Alermanager). API that server Grafana Alerting is protected with folder scope. External Providers - datasource scope.

Reply:
I see. I think we currently don't have any other cases where we store two different scopes for the same resource in the DB yet, but hopefully it won't cause issues 🤞